lib/logflare/application.ex

```diff
@@ -90,89 +81,136 @@ defmodule Logflare.Application do
+        Logs.RejectedLogEvents,
+        # init Counters before Supervisof as Supervisor calls Counters through table create
+        Sources.Counters,
+        Sources.RateCounters,
+        PubSubRates.Rates,
+        PubSubRates.Buffers,
+        PubSubRates.Inserts,
+        Logflare.Source.Supervisor,
 
-      # Startup tasks
-      {Task, fn -> startup_tasks() end},
+        # If we get a log event and the Source.Supervisor is not up it will 500
+        LogflareWeb.Endpoint,
+        {GRPC.Server.Supervisor, {LogflareGrpc.Endpoint, grpc_port, cred: grpc_creds}},
+        # Monitor system level metrics
+        Logflare.SystemMetricsSup,
 
-      # v2 ingestion pipelines
-      {DynamicSupervisor, strategy: :one_for_one, name: Logflare.Backends.SourcesSup},
-      {DynamicSupervisor, strategy: :one_for_one, name: Logflare.Backends.RecentLogsSup},
-      {DynamicSupervisor,
-       strategy: :one_for_one, name: Logflare.Backends.Adaptor.PostgresAdaptor.Supervisor},
-      {DynamicSupervisor,
-       strategy: :one_for_one, name: Logflare.Backends.Adaptor.PostgresAdaptor.PgRepoSupervisor},
-      {Registry,
-       name: Logflare.Backends.SourceRegistry,
-       keys: :unique,
-       partitions: System.schedulers_online()},
-      {Registry, name: Logflare.Backends.SourceDispatcher, keys: :duplicate},
+        # For Logflare Endpoints
+        {DynamicSupervisor, strategy: :one_for_one, name: Logflare.Endpoints.Cache},
 
-      # citrine scheduler for alerts
-      Logflare.AlertsScheduler
-    ] ++ conditional_children() ++ common_children()
+        # Startup tasks
+        {Task, fn -> startup_tasks() end},
+
+        # v2 ingestion pipelines
+        {DynamicSupervisor, strategy: :one_for_one, name: Logflare.Backends.SourcesSup},
+        {DynamicSupervisor, strategy: :one_for_one, name: Logflare.Backends.RecentLogsSup},
+        {DynamicSupervisor,
+         strategy: :one_for_one, name: Logflare.Backends.Adaptor.PostgresAdaptor.Supervisor},
+        {DynamicSupervisor,
+         strategy: :one_for_one, name: Logflare.Backends.Adaptor.PostgresAdaptor.PgRepoSupervisor},
+        {Registry,
+         name: Logflare.Backends.SourceRegistry,
+         keys: :unique,
+         partitions: System.schedulers_online()},
+        {Registry, name: Logflare.Backends.SourceDispatcher, keys: :duplicate},
+
+        # citrine scheduler for alerts
+        Logflare.AlertsScheduler
+      ] ++ common_children()
   end
 
   def conditional_children do
     goth =
       case Application.get_env(:goth, :json) do
-        nil -> []
-        _ -> [get_goth_child_spec()]
+        nil ->
+          []
+
+        json ->
+          # Setup Goth for GCP connections
+          credentials = Jason.decode!(json)
+          scopes = ["https://www.googleapis.com/auth/cloud-platform"]
+          source = {:service_account, credentials, scopes: scopes}
+
+          spec =
```

Comment:
I think we need to tell it to specificaly use our `Logflare.FinchDefault` pool?

https://github.com/peburrows/goth/blob/master/lib/goth/token.ex#L209C13-L209C13de

Reply:
done